pipeline/src/pipeline.rs

```diff
@@ -261,7 +263,15 @@ impl<T: FieldElement> Pipeline<T> {
         self
     }
 
-    pub fn with_prover_inputs(self, inputs: Vec<T>) -> Self {
+    pub fn add_data<S: serde::Serialize + Send + Sync + 'static>(
+        self,
+        channel: u32,
+        data: &S,
+    ) -> Self {
+        self.add_query_callback(Box::new(serde_data_to_query_callback(channel, data)))
+    }
+
+    pub fn with_prover_inputs(self, inputs: HashMap<u32, Vec<T>>) -> Self {
```

Comment:
I think in practice we only use this with maps with one key (0), right? In that case, maybe we can add a convenience function of the old signature to use in tests?

Reply:
Right